composer/models/base.py

```diff
@@ -58,39 +146,72 @@ def metrics(self, train: bool = False) -> Metrics:
 
         Args:
             train (bool, optional): True to return metrics that should be computed
-                during training and False otherwise. (default: ``False``)
+                during training and False otherwise. (default: ``False``). This flag is set automatically by the
+                :class:`~composer.trainer.trainer.Trainer`
 
         Returns:
-            Metrics: A ``Metrics`` object.
+             Metric or MetricCollection: An instance of :class:`~torchmetrics.Metric` or :class:`~torchmetrics.MetricCollection`.
         """
         raise NotImplementedError('Implement metrics in your ComposerModel to run validation.')
 
     def validate(self, batch: Batch) -> Tuple[Any, Any]:
-        """Compute model outputs on provided data.
+        """Compute model outputs on provided data. Will be called by the trainer with :func:`torch.nograd()` enabled.
 
         The output of this function will be directly used as input
         to all metrics returned by :meth:`metrics`.
 
+        Example:
+
+        .. code-block:: python
+
+            def validate(self, batch): # batch is the output of the dataloader
+                inputs, targets = batch
+                outputs = self.model(inputs)
+                return outputs, targets # return a tuple of (``outputs``, ``targets``)
+
+
+        How validate outputs are passed to :meth:`metrics` (pseudocode):
+
+        .. code-block:: python
+
+            metrics = model.metrics(train=False) # get torchmetrics
+
+            for batch in val_dataloader:
+                outputs, targets = model.validate(batch)
+                metrics.update(outputs, targets)  # update metrics with output, targets for each batch
+
+            metrics.compute() # compute final metrics
+
         Args:
-            batch (Batch): The data to perform validation with.
-                Specified as a tuple of tensors (input, target).
+            batch (~composer.core.types.Batch): The output batch from dataloader
 
         Returns:
-            Tuple[Any, Any]: Tuple that is passed directly to the
-                `update()` methods of the metrics returned by :meth:`metrics`.
-                Most often, this will be a tuple of the form (predictions, targets).
+            Tuple[Any, Any]: A Tuple of (``outputs``, ``targets``) that is passed directly to the
+                :meth:`~torchmetrics.Metric.update` methods of the metrics returned by :meth:`metrics`.
         """
         raise NotImplementedError('Implement validate in your ComposerModel to run validation.')
 
 
 class ComposerClassifier(ComposerModel):
-    """Implements the base logic that all classifiers can build on top of.
+    """A convenience class that creates a :class:`~composer.models.base.ComposerModel` for classification tasks from a
```

Comment:
here and throughout should use the global scope `.ComposerModel` and `.ComposerClassifier` to make the docstrings more less verbose.